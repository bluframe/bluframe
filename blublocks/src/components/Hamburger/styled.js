// @flow

import styled, { type StyledComponent } from "styled-components"
import type { Theme } from "@bluframe/blublocks"

export const Bar: StyledComponent<{||}, Theme, HTMLSpanElement> = styled.span`
  background: ${({ theme }) => theme.palette.secondary.main};
  display: block;
  width: 24px;
  height: 4px;
  border-radius: 2px;

  + span {
    margin-top: 4px;
  }
`

export const Button: StyledComponent<
  {||},
  Theme,
  HTMLButtonElement
> = styled.button`
  position: relative;
  float: right;
  padding: 9px 10px;
  margin-top: 18px;
  margin-right: 0;
  margin-bottom: 8px;
  background-color: transparent;
  background-image: none;
  border: none;
  border-radius: 4px;

  &:focus {
    background-color: white;
    outline: none;
  }
`

export const ScreenReaderOnly: StyledComponent<
  any,
  mixed,
  HTMLSpanElement
> = styled.span`
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0;
`
