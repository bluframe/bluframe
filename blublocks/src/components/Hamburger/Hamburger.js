// @flow

import { Bar, Button, ScreenReaderOnly } from "./styled"
import type { ComponentProps } from "."
import React from "react"

const Hamburger = ({ onClick }: ComponentProps): React$Node => (
  <Button onClick={onClick} type="button">
    <ScreenReaderOnly>{"Toggle navigation"}</ScreenReaderOnly>

    <Bar />
    <Bar />
    <Bar />
  </Button>
)

export default Hamburger
