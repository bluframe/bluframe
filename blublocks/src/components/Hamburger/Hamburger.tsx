import { Bar, Button, ScreenReaderOnly } from "./styled"
import type { ComponentProps } from "."
import React from "react"

const Hamburger = ({
  onClick,
  isMenuExpanded = false,
  menuId
}: ComponentProps) => (
  <Button
    aria-controls={menuId}
    aria-expanded={isMenuExpanded}
    aria-label="Toggle navigation"
    onClick={onClick}
    role="menu"
    type="button"
  >
    <ScreenReaderOnly>{"Toggle navigation"}</ScreenReaderOnly>

    <Bar />
    <Bar />
    <Bar />
  </Button>
)

export default Hamburger
