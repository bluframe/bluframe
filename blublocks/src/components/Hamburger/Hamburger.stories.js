// @flow

import Hamburger, { type ComponentProps } from "components/Hamburger"
import React from "react"

const stories = {
  argTypes: {
    onClick: { action: "button click" }
  },
  component: Hamburger,
  title: "Components/Hamburger"
}

const Template = (args: ComponentProps) => <Hamburger {...args} />

export const Default: any = Template.bind({})

export default stories
