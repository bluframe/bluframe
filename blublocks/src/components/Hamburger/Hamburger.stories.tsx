import Hamburger, { Props } from "components/Hamburger"
import React from "react"

const stories = {
  argTypes: {
    onClick: { action: "button click" }
  },
  component: Hamburger,
  title: "Components/Hamburger"
}

const Template = (args: Props) => <Hamburger {...args} />

export const Default: any = Template.bind({})

export default stories
