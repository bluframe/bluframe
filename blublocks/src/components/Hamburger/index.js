// @flow

import Hamburger from "./Hamburger"
import type { HamburgerProps } from "@bluframe/blublocks"

export type ComponentProps = {|
  ...HamburgerProps
|}

export default Hamburger
