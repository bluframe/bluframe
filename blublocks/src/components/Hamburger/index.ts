import Hamburger from "./Hamburger"

export interface Props {
  isMenuExpanded?: boolean
  menuId?: string
  onClick?: () => void
}

export interface ComponentProps extends Props {}

export default Hamburger
