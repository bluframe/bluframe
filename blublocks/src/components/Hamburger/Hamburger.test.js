// @flow

import Hamburger from "./Hamburger.js"
import React from "react"
import { render } from "tests"

const props = {
  onClick: jest.fn()
}

describe("Hamburger", () => {
  it("renders", () => {
    const { container } = render(<Hamburger {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
