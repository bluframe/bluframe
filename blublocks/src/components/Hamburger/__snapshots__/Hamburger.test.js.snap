// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger renders 1`] = `
<button
  aria-controls="main-navigation"
  aria-expanded="false"
  aria-label="Toggle navigation"
  class="styled__Button-sc-45lonn-1 kASBxd"
  role="menu"
  type="button"
>
  <span
    class="styled__ScreenReaderOnly-sc-45lonn-2 iQoRVS"
  >
    Toggle navigation
  </span>
  <span
    class="styled__Bar-sc-45lonn-0 kUYBpF"
  />
  <span
    class="styled__Bar-sc-45lonn-0 kUYBpF"
  />
  <span
    class="styled__Bar-sc-45lonn-0 kUYBpF"
  />
</button>
`;
